src/algorithm.rs

```diff
@@ -34,7 +36,6 @@ pub(crate) struct Algorithm<N: Network + 'static, K: KVStore, A: App<K> + 'stati
     pacemaker: Pacemaker<N>,
     block_sync_client: BlockSyncClient<N>,
     shutdown_signal: Receiver<()>,
-    event_publisher: Option<Sender<Event>>,
 }
 
 impl<N: Network + 'static, K: KVStore, A: App<K> + 'static> Algorithm<N, K, A> {
```

Comment:
Nitpick: the order of the arguments of the `new` method could be made more logical. For example, `progress_msg_buffer_capacity` should probably sit right after `progress_msg_receiver`